### DISCUSSION TITLE
ConfigDicts aren't converted to dataclasses when specified in a list

### DISCUSSION THREAD

**Discussion by DirkKuhn** [mit-ll-responsible-ai/hydra-zen#441] [Q&A] [answered]
The conversion back to dataclasses doesn't seem to work, when they are specified in a list.
Here is a minimal example.
```python
from collections.abc import Sequence
from dataclasses import dataclass
from hydra_zen import store, zen, builds
from hydra_zen.typing import ZenConvert

@dataclass
class B:
    x: int

def app(dc: Sequence[B]) -> None:
    assert all(isinstance(b, B) for b in dc)

Config = builds(
    app,
    populate_full_signature=True,
    dc=[B(x=0)],
    zen_convert=ZenConvert(dataclass=True),
    hydra_convert='all'
)
store(Config, name="config")

if __name__ == "__main__":
    store.add_to_hydra_store()
    zen(app).hydra_main(config_path=None, config_name="config", version_base="1.3")
```
It fails with an assertion error. The entry in `dc` is a `DictConfig`.
Do I use `zen_convert` and `hydra_convert` the right way?
So far I reverted back to normal classes from dataclasses to solve this issue.

Thanks!

**Comment by rsokl** [comment, marked as answer]
Could you repost this as an issue? It sounds like it is a bug that should be fixed

**Reply by DirkKuhn** [reply]
Done :)

**Reply by rsokl** [reply]
https://github.com/mit-ll-responsible-ai/hydra-zen/issues/442
